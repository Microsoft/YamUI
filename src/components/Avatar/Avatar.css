/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
@import "src/css/variables/colors.css";

.y-avatar {
  position: relative;
}

.y-avatar__size-xLarge {
  height: 72px;
  width: 72px;
}
.y-avatar__size-large {
  height: 48px;
  width: 48px;
}
.y-avatar__size-medium {
  height: 40px;
  width: 40px;
}
.y-avatar__size-small {
  height: 32px;
  width: 32px;
}
.y-avatar__size-xSmall {
  height: 24px;
  width: 24px;
}

/* Fabric Persona adds extra height here. This should be unnecessary
    if we can extract a Fabric Avatar component out of Persona. */
.y-avatar__size-xLarge .ms-Persona {
  height: 72px;
}
.y-avatar__size-large .ms-Persona {
  height: 48px;
}
.y-avatar__size-medium .ms-Persona {
  height: 40px;
}
.y-avatar__size-small .ms-Persona {
  height: 32px;
}
.y-avatar__size-xSmall .ms-Persona {
  height: 24px;
}

.y-avatar--badge {
  position: absolute;
  height: 20px;
  width: 20px;
  bottom: -2px;
  right: -2px;
  border-radius: 50%;
  border: solid 2px #fff;
  background: #fff;
  color: $color__river;
}
.y-avatar__size-small--badge,
.y-avatar__size-xSmall--badge {
  height: 16px;
  width: 16px;
}

.y-avatar--badge > * {
  top: 0 !important;
}

/* Fabric CSS overrides */
 .y-avatar > .ms-Persona {
   margin: 0;
 }
.y-avatar__borderType-soft .ms-Persona-imageArea,
.y-avatar__borderType-soft .ms-Image {
  border-radius: 2px;
}
