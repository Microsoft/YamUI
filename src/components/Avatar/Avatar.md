### Notes for use

`Avatar` takes care of setting the alt text for you when you pass the one required attribute, `name`.

### Examples

Without image:

```js { "props": { "data-description": "without image" } }
const { AvatarSize } = require('.');

<LayoutList direction="horizontal">
  <LayoutListItem>
    <Avatar size={AvatarSize.XLARGE} name="Xtra Large" />
  </LayoutListItem>
  <LayoutListItem>
    <Avatar size={AvatarSize.LARGE} name="Large" />
  </LayoutListItem>
  <LayoutListItem>
    <Avatar size={AvatarSize.MEDIUM} name="Medium" />
  </LayoutListItem>
  <LayoutListItem>
    <Avatar size={AvatarSize.SMALL} name="Small" />
  </LayoutListItem>
  <LayoutListItem>
    <Avatar size={AvatarSize.XSMALL} name="Xtra Small" />
  </LayoutListItem>
</LayoutList>
```

With image:

```js { "props": { "data-description": "with image" } }
const { AvatarSize } = require('.');

<LayoutList direction="horizontal">
  <LayoutListItem>
    <Avatar size={AvatarSize.XLARGE} name={user.name} imageUrl={user.imageUrl} />
  </LayoutListItem>
  <LayoutListItem>
    <Avatar size={AvatarSize.LARGE} name={user.name} imageUrl={user.imageUrl} />
  </LayoutListItem>
  <LayoutListItem>
    <Avatar size={AvatarSize.MEDIUM} name={user.name} imageUrl={user.imageUrl} />
  </LayoutListItem>
  <LayoutListItem>
    <Avatar size={AvatarSize.SMALL} name={user.name} imageUrl={user.imageUrl} />
  </LayoutListItem>
  <LayoutListItem>
    <Avatar size={AvatarSize.XSMALL} name={user.name} imageUrl={user.imageUrl} />
  </LayoutListItem>
</LayoutList>
```

With image and badge:

```js { "props": { "data-description": "with image and badge" } }
const { AvatarSize } = require('.');
const AdminBadge = require('../Icon/icons/BadgeAdmin').default;

const badgeContent = (
  <AdminBadge block={true} />
);

<LayoutList direction="horizontal">
  <LayoutListItem>
    <Avatar
      size={AvatarSize.XLARGE}
      name={user.name}
      imageUrl={user.imageUrl}
      badgeContent={badgeContent}
      badgeDescription="Admin"
    />
  </LayoutListItem>
  <LayoutListItem>
    <Avatar
      size={AvatarSize.LARGE}
      name={user.name}
      imageUrl={user.imageUrl}
      badgeContent={badgeContent}
      badgeDescription="Admin"
    />
  </LayoutListItem>
  <LayoutListItem>
    <Avatar
      size={AvatarSize.MEDIUM}
      name={user.name}
      imageUrl={user.imageUrl}
      badgeContent={badgeContent}
      badgeDescription="Admin"
    />
  </LayoutListItem>
  <LayoutListItem>
    <Avatar
      size={AvatarSize.SMALL}
      name={user.name}
      imageUrl={user.imageUrl}
      badgeContent={badgeContent}
      badgeDescription="Admin"
    />
  </LayoutListItem>
  <LayoutListItem>
    <Avatar
      size={AvatarSize.XSMALL}
      name={user.name}
      imageUrl={user.imageUrl}
      badgeContent={badgeContent}
      badgeDescription="Admin"
    />
  </LayoutListItem>
</LayoutList>
```

With image and soft border:

```js { "props": { "data-description": "with image and soft border" } }
const { BorderType, AvatarSize } = require('.');

<LayoutList direction="horizontal">
  <LayoutListItem>
    <Avatar
      size={AvatarSize.XLARGE}
      name={group.name}
      imageUrl={group.imageUrl}
      borderType={BorderType.SOFT}
    />
  </LayoutListItem>
  <LayoutListItem>
    <Avatar
      size={AvatarSize.LARGE}
      name={group.name}
      imageUrl={group.imageUrl}
      borderType={BorderType.SOFT}
    />
  </LayoutListItem>
  <LayoutListItem>
    <Avatar
      size={AvatarSize.MEDIUM}
      name={group.name}
      imageUrl={group.imageUrl}
      borderType={BorderType.SOFT}
    />
  </LayoutListItem>
  <LayoutListItem>
    <Avatar
      size={AvatarSize.SMALL}
      name={group.name}
      imageUrl={group.imageUrl}
      borderType={BorderType.SOFT}
    />
  </LayoutListItem>
  <LayoutListItem>
    <Avatar
      size={AvatarSize.XSMALL}
      name={group.name}
      imageUrl={group.imageUrl}
      borderType={BorderType.SOFT}
    />
  </LayoutListItem>
</LayoutList>
```

Example colors:

```js { "props": { "data-description": "colors" } }
const { GutterSize } = require('../Block');
const { AvatarSize } = require('.');

<div>
  <Block bottomSpacing={GutterSize.SMALL}>
    A background color will be automatically set when an <em>imageUrl</em> is not provided. Fabric handles this behind the scenes for us, based on the <em>name</em> prop passed in.
  </Block>
  <FixedGridRow gutterSize={GutterSize.SMALL} bottomSpacing={GutterSize.SMALL}>
    <FixedGridColumn>
      <Avatar size={AvatarSize.MEDIUM} name="Richie Black" />
    </FixedGridColumn>
    <FixedGridColumn>
      <Avatar size={AvatarSize.MEDIUM} name="Sara Michaels" />
    </FixedGridColumn>
    <FixedGridColumn>
      <Avatar size={AvatarSize.MEDIUM} name="Charles Blue" />
    </FixedGridColumn>
    <FixedGridColumn>
      <Avatar size={AvatarSize.MEDIUM} name="Billy Jean" />
    </FixedGridColumn>
    <FixedGridColumn>
      <Avatar size={AvatarSize.MEDIUM} name="James Woods" />
    </FixedGridColumn>
    <FixedGridColumn>
      <Avatar size={AvatarSize.MEDIUM} name="John Babcock" />
    </FixedGridColumn>
    <FixedGridColumn>
      <Avatar size={AvatarSize.MEDIUM} name="Perry McDouglas" />
    </FixedGridColumn>
  </FixedGridRow>
  <FixedGridRow gutterSize={GutterSize.SMALL} bottomSpacing={GutterSize.SMALL}>
    <FixedGridColumn>
      <Avatar size={AvatarSize.MEDIUM} name="Monte Luther" />
    </FixedGridColumn>
    <FixedGridColumn>
      <Avatar size={AvatarSize.MEDIUM} name="Zoe Reynolds" />
    </FixedGridColumn>
    <FixedGridColumn>
      <Avatar size={AvatarSize.MEDIUM} name="Jose Monahan" />
    </FixedGridColumn>
    <FixedGridColumn>
      <Avatar size={AvatarSize.MEDIUM} name="Sean Johnson" />
    </FixedGridColumn>
    <FixedGridColumn>
      <Avatar size={AvatarSize.MEDIUM} name="Toby Edward" />
    </FixedGridColumn>
    <FixedGridColumn>
      <Avatar size={AvatarSize.MEDIUM} name="Willie Nelson" />
    </FixedGridColumn>
    <FixedGridColumn>
      <Avatar size={AvatarSize.MEDIUM} name="Joe Johnson" />
    </FixedGridColumn>
  </FixedGridRow>
</div>
```