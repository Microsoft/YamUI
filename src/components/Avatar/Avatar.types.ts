/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import { CustomizableComponentProps } from '../Customizer';
import { BorderType } from '../Image/types';
export { BorderType };

export enum AvatarSize {
  /**
   * 72px.
   */
  XLARGE = 'xLarge',
  /**
   * 48px.
   */
  LARGE = 'large',
  /**
   * 40px;
   */
  MEDIUM = 'medium',
  /**
   * 32px.
   */
  SMALL = 'small',
  /**
   * 24px.
   */
  XSMALL = 'xSmall',
}

export interface AvatarProps extends CustomizableComponentProps {
  /**
   * Name of the person or object being represented. Will be used as accessible alt text.
   */
  name: string;

  /**
   * Element to be used as badge. You can set its height and width to fill the available area.
   */
  badgeContent?: React.ReactNode;

  /**
   * Short accessible description of the badge. Will be appended to name if provided.
   */
  badgeDescription?: string;

  /**
   * Type of border around the avatar.
   * @default BorderType.ROUND
   */
  borderType?: BorderType;

  /**
   * Image source URL.
   */
  imageUrl?: string;

  /**
   * XLARGE: 72px, LARGE: 48px, MEDIUM: 40px, SMALL: 32px, XSMALL: 24px.
   * @default AvatarSize.MEDIUM
   */
  size?: AvatarSize;

  /**
   * Will hide the image until it has loaded, then fade it in.
   * @default false
   */
  imageShouldFadeIn?: boolean;
}
