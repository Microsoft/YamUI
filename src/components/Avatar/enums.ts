/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
export enum AvatarBorderType{
  SOFT = 'soft',
  ROUND = 'round',
}

export enum AvatarSize {
  XLARGE = 'xLarge',
  LARGE = 'large',
  MEDIUM = 'medium',
  SMALL = 'small',
  XSMALL = 'xSmall',
}
