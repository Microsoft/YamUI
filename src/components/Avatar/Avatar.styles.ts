/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import { mergeStyleSets, ITheme } from 'office-ui-fabric-react/lib/Styling';
import { memoizeFunction } from 'office-ui-fabric-react/lib/Utilities';
import { IPersonaCoinStyles } from 'office-ui-fabric-react/lib/PersonaCoin';
import { AvatarSize, BorderType } from './Avatar.types';
import { borders } from '../../util/styles/borders';

export interface AvatarStyleProps {
  borderType: BorderType;
  size: AvatarSize;
}

export const getPersonaCoinStyles = memoizeFunction(
  (props: AvatarStyleProps): (() => Partial<IPersonaCoinStyles>) => {
    const { borderType, size } = props;
    const borderRadius = borders[borderType];
    const initials = size === AvatarSize.XLARGE ? { lineHeight: '70px' } : {};
    return () => ({
      initials,
      image: {
        borderRadius,
      },
      imageArea: {
        borderRadius,
      },
    });
  },
);

export interface AvatarClassNameProps {
  size: AvatarSize;
  theme: ITheme;
}

export const getClassNames = memoizeFunction((props: AvatarClassNameProps) => {
  const { size, theme } = props;
  return mergeStyleSets({
    root: {
      position: 'relative',
    },
    badge: {
      position: 'absolute',
      height: size === AvatarSize.SMALL || size === AvatarSize.XSMALL ? '16px' : '20px',
      width: size === AvatarSize.SMALL || size === AvatarSize.XSMALL ? '16px' : '20px',
      bottom: '-2px',
      right: '-2px',
      borderRadius: borders.round,
      border: `solid 2px ${theme.palette.white}`,
      background: theme.palette.white,
      color: theme.palette.themeDark,
      selectors: {
        /* Remove when this is addressed: https://github.com/Microsoft/YamUI/issues/327 */
        '> .y-icon': {
          top: 0,
        },
      },
    },
  });
});
