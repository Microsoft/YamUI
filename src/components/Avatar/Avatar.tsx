/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import '../../yamui';
import * as React from 'react';
import { PersonaCoin, PersonaSize } from 'office-ui-fabric-react/lib/PersonaCoin';
import { CustomizableComponentProps, nullTheme } from '../Customizer';
import ScreenreaderText from '../ScreenreaderText';
import { AvatarSize } from './types';
import { BorderType } from '../Image/types';
import { getPersonaCoinStyles, getClassNames } from './Avatar.styles';
import { customizable } from '@uifabric/utilities';
import { ITheme } from '@uifabric/styling';

export { BorderType, AvatarSize };

const SizeMap = {
  [AvatarSize.XLARGE]: PersonaSize.size72,
  [AvatarSize.LARGE]: PersonaSize.size48,
  [AvatarSize.MEDIUM]: PersonaSize.size40,
  [AvatarSize.SMALL]: PersonaSize.size32,
  [AvatarSize.XSMALL]: PersonaSize.size24,
};

export interface AvatarProps extends CustomizableComponentProps {
  /**
   * Name of the person or object being represented. Will be used as accessible alt text.
   */
  name: string;

  /**
   * Element to be used as badge. You can set its height and width to fill the available area.
   */
  badgeContent?: React.ReactNode;

  /**
   * Short accessible description of the badge. Will be appended to name if provided.
   */
  badgeDescription?: string;

  /**
   * Type of border around the avatar.
   * @default BorderType.ROUND
   */
  borderType?: BorderType;

  /**
   * Image source URL.
   */
  imageUrl?: string;

  /**
   * XLARGE: 72px, LARGE: 48px, MEDIUM: 40px, SMALL: 32px, XSMALL: 24px.
   * @default AvatarSize.MEDIUM
   */
  size?: AvatarSize;

  /**
   * Will hide the image until it has loaded, then fade it in.
   * @default false
   */
  imageShouldFadeIn?: boolean;
}

/**
 * An `Avatar` shows a thumbnail representation of both an individual or group.
 */
export class Avatar extends React.Component<AvatarProps> {
  public render() {
    const {
      badgeContent,
      imageUrl,
      name,
      badgeDescription,
      theme = nullTheme,
      size = AvatarSize.MEDIUM,
      imageShouldFadeIn = false,
      borderType = BorderType.ROUND,
    } = this.props;
    const classNames = getClassNames({ size, theme: theme as ITheme });
    const badge = badgeContent && <div className={classNames.badge}>{badgeContent}</div>;

    return (
      <div className={`y-avatar ${classNames.root}`}>
        <PersonaCoin
          imageUrl={imageUrl}
          size={SizeMap[size]}
          hidePersonaDetails={true}
          text={name}
          imageShouldFadeIn={imageShouldFadeIn}
          styles={getPersonaCoinStyles({ borderType, size })}
        />
        {badge}
        <ScreenreaderText>{badgeDescription ? `${name} - ${badgeDescription}` : name}</ScreenreaderText>
      </div>
    );
  }
}

@customizable('Avatar', ['theme'])
export default class CustomizableAvatar extends Avatar {}
