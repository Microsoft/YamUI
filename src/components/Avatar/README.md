# Avatar

An `Avatar` shows a thumbnail representation of both an individual or group.

## Properties

| Prop | Type | Description |
| ---- | ---- | ----------- |
| size | AvatarSize | Size of avatar. |
| name * | string | Name of the person or object being represented. Will be used as accessible alt text. |
| initials | string | Initials of person to display if no mugshot provided. |
| imageUrl | string | Image source URL. |
| borderType | BorderType | Round or soft border. Defaults to round. |
| badgeContent | ReactNode | Image, SVG, icon, etc. See example below. |
| badgeDescription | string | A short accessible description of the badge. Will be appended to name if provided. |

\* property is required

## Notes for use

`Avatar` takes care of setting the alt text for you when you pass the one required attribute, `name`.

You must provide either an `imageUrl` property or an `initials` property.

## Examples

### Person avatar without image

```js
<Avatar
  size={AvatarSize.XLARGE}
  name="Walter White"
  initials="WW"
/>
```

### Group avatar with image

```js
<Avatar
  size={AvatarSize.XSMALL}
  name="Marketing Team"
  imageUrl={antelope}
  borderType={BorderType.SOFT}
/>
```

## Avatar with image and badge

```js
import Avatar from 'yamui/dist/components/Avatar';
import Image from 'yamui/dist/components/Image';

const badgeImage = <Image source="badge-image.png" description="" fullWidth={true} />;

<Avatar
  size={AvatarSize.XSMALL}
  name="Smiley Face"
  imageUrl="my-image.jpg"
  badgeContent={badgeImage}
  badgeDescription="Admin"
/>
```
