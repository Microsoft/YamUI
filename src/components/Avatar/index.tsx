/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import '../../yamui';
import * as React from 'react';
import { Persona, PersonaSize } from 'office-ui-fabric-react/lib/Persona';
import { BaseComponentProps } from '../../util/BaseComponent/props';
import ScreenreaderText from '../ScreenreaderText';
import { AvatarBorderType, AvatarSize } from './enums';
import './avatar.css';

export { AvatarBorderType, AvatarSize };

const SizeMap = {
  [AvatarSize.XLARGE]: PersonaSize.large,
  [AvatarSize.LARGE]: PersonaSize.regular,
  [AvatarSize.MEDIUM]: PersonaSize.small,
  [AvatarSize.SMALL]: PersonaSize.extraSmall,
  [AvatarSize.XSMALL]: PersonaSize.extraExtraSmall,
};

export interface BaseAvatarProps extends BaseComponentProps {
  /**
   * Image source URL.
   */
  imageUrl?: string;

  /**
   * 2 letters to be displayed if an imageUrl wasn't provided.
   */
  initials?: string;

  /**
   * The name of the person or object being represented. Will be used as accessible alt text.
   */
  name: string;

  /**
   * Image, SVG, icon, etc. You can set its height and width to fill the available area.
   */
  badgeContent?: React.ReactNode;

  /**
   * A short accessible description of the badge. Will be appended to name if provided.
   */
  badgeDescription?: string;

  /**
   * Round or soft border. Defaults to round.
   */
  borderType?: AvatarBorderType;

  /**
   * XLARGE: 72px, LARGE: 48px, MEDIUM: 40px, SMALL: 32px, XSMALL: 24px. Defaults to medium.
   */
  size?: AvatarSize;
}
export interface ImageAvatarProps extends BaseAvatarProps {
  /**
   * Image source URL.
   */
  imageUrl: string;
}
export interface InitialsAvatarProps extends BaseAvatarProps {
  /**
   * 2 letters to be displayed if an imageUrl wasn't provided.
   */
  initials: string;
}

// AvatarProps requires either imageUrl OR initials
export type AvatarProps = ImageAvatarProps | InitialsAvatarProps;

export default class Avatar extends React.PureComponent<AvatarProps, {}> {
  static defaultProps: Partial<AvatarProps> = {
    borderType: AvatarBorderType.ROUND,
    size: AvatarSize.MEDIUM,
  };

  render() {
    const personaSize = SizeMap[this.props.size as string];

    const badge = this.props.badgeContent && (
      <div className={`y-avatar--badge y-avatar__size-${this.props.size}--badge`}>
        {this.props.badgeContent}
      </div>
    );

    return (
      <div className={this.getClasses()}>
        <Persona
          imageUrl={this.props.imageUrl}
          imageInitials={this.getInitials()}
          size={personaSize}
          hidePersonaDetails={true}
          primaryText={this.props.name}
        />
        {badge}
        <ScreenreaderText>{this.getAccessibleText()}</ScreenreaderText>
      </div>
    );
  }

  private getInitials() {
    if (!this.props.initials) {
      return;
    }

    if (this.props.size === AvatarSize.XSMALL) {
      return this.props.initials.substring(0, 1);
    } else {
      return this.props.initials.substring(0, 2);
    }
  }

  private getAccessibleText() {
    const { name, badgeDescription } = this.props;
    if (badgeDescription) {
      return `${name} - ${badgeDescription}`;
    }
    return name;
  }

  private getClasses() {
    const classes: string[] = ['y-avatar', `y-avatar__size-${this.props.size}`];
    if (this.props.borderType !== Avatar.defaultProps.borderType) {
      classes.push(`y-avatar__borderType-${this.props.borderType}`);
    }
    if (this.props.className) {
      classes.push(this.props.className);
    }
    return classes.join(' ');
  }
}
