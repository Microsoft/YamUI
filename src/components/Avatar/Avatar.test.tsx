/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import * as React from 'react';
import { shallow, mount, ShallowWrapper } from 'enzyme';
import Text from '../Text';
import CustomizableAvatar, { Avatar } from './Avatar';
import { AvatarProps, AvatarSize, BorderType } from './Avatar.types';
import Customizer, { defaultTheme } from '../Customizer';

describe('<Avatar />', () => {
  let component: ShallowWrapper<AvatarProps>;

  describe('with image url', () => {
    beforeEach(() => {
      component = shallow(<Avatar name="NAME" imageUrl="test.jpg" />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with too many characters', () => {
    beforeEach(() => {
      component = shallow(<Avatar name="First Last" />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with imageShouldFadeIn', () => {
    beforeEach(() => {
      component = shallow(<Avatar name="NAME" imageShouldFadeIn={true} />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with badge content', () => {
    beforeEach(() => {
      component = shallow(
        <Avatar name="NAME" imageUrl="test.jpg" size={AvatarSize.XSMALL} badgeContent={<Text>badge</Text>} />,
      );
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with soft border type', () => {
    beforeEach(() => {
      component = shallow(<Avatar name="NAME" imageUrl="test.jpg" borderType={BorderType.SOFT} />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with size', () => {
    beforeEach(() => {
      component = shallow(<Avatar name="NAME" imageUrl="test.jpg" size={AvatarSize.XLARGE} />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with additional className', () => {
    beforeEach(() => {
      component = shallow(<Avatar name="NAME" imageUrl="test.jpg" className="TEST_CLASSNAME" />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('accessible text', () => {
    describe('without badge description', () => {
      beforeEach(() => {
        component = shallow(<Avatar name="NAME" imageUrl="test.jpg" />);
      });

      it('matches its snapshot', () => {
        expect(component).toMatchSnapshot();
      });
    });

    describe('with badge description', () => {
      beforeEach(() => {
        component = shallow(<Avatar name="NAME" imageUrl="test.jpg" badgeDescription="BADGE" />);
      });

      it('matches its snapshot', () => {
        expect(component).toMatchSnapshot();
      });
    });
  });

  describe('with customizer', () => {
    let mountedComponent: ReactWrapper;
    const theme = defaultTheme;

    beforeEach(() => {
      mountedComponent = mount(
        <Customizer settings={{ theme }}>
          <CustomizableAvatar name="NAME" />
        </Customizer>,
      );
    });

    it('receives custom theme', () => {
      expect(mountedComponent.find('CustomizableAvatar').prop('theme')).toBe(theme);
    });
  });
});
