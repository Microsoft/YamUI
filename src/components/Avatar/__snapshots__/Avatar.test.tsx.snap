// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> accessible text with badge description matches its snapshot 1`] = `
<div
  className="y-avatar y-avatar__size-medium"
>
  <StyledCustomizedPersonaCoin
    hidePersonaDetails={true}
    imageShouldFadeIn={false}
    imageUrl="test.jpg"
    size={12}
    text="NAME"
  />
  <ScreenreaderText>
    NAME - BADGE
  </ScreenreaderText>
</div>
`;

exports[`<Avatar /> accessible text without badge description matches its snapshot 1`] = `
<div
  className="y-avatar y-avatar__size-medium"
>
  <StyledCustomizedPersonaCoin
    hidePersonaDetails={true}
    imageShouldFadeIn={false}
    imageUrl="test.jpg"
    size={12}
    text="NAME"
  />
  <ScreenreaderText>
    NAME
  </ScreenreaderText>
</div>
`;

exports[`<Avatar /> with additional className matches its snapshot 1`] = `
<div
  className="y-avatar y-avatar__size-medium TEST_CLASSNAME"
>
  <StyledCustomizedPersonaCoin
    hidePersonaDetails={true}
    imageShouldFadeIn={false}
    imageUrl="test.jpg"
    size={12}
    text="NAME"
  />
  <ScreenreaderText>
    NAME
  </ScreenreaderText>
</div>
`;

exports[`<Avatar /> with badge content matches its snapshot 1`] = `
<div
  className="y-avatar y-avatar__size-xSmall"
>
  <StyledCustomizedPersonaCoin
    hidePersonaDetails={true}
    imageShouldFadeIn={false}
    imageUrl="test.jpg"
    size={10}
    text="NAME"
  />
  <div
    className="y-avatar--badge y-avatar__size-xSmall--badge"
  >
    <Component>
      badge
    </Component>
  </div>
  <ScreenreaderText>
    NAME
  </ScreenreaderText>
</div>
`;

exports[`<Avatar /> with image url matches its snapshot 1`] = `
<div
  className="y-avatar y-avatar__size-medium"
>
  <StyledCustomizedPersonaCoin
    hidePersonaDetails={true}
    imageShouldFadeIn={false}
    imageUrl="test.jpg"
    size={12}
    text="NAME"
  />
  <ScreenreaderText>
    NAME
  </ScreenreaderText>
</div>
`;

exports[`<Avatar /> with size matches its snapshot 1`] = `
<div
  className="y-avatar y-avatar__size-xLarge"
>
  <StyledCustomizedPersonaCoin
    hidePersonaDetails={true}
    imageShouldFadeIn={false}
    imageUrl="test.jpg"
    size={14}
    text="NAME"
  />
  <ScreenreaderText>
    NAME
  </ScreenreaderText>
</div>
`;

exports[`<Avatar /> with soft border type matches its snapshot 1`] = `
<div
  className="y-avatar y-avatar__size-medium y-avatar__borderType-soft"
>
  <StyledCustomizedPersonaCoin
    hidePersonaDetails={true}
    imageShouldFadeIn={false}
    imageUrl="test.jpg"
    size={12}
    text="NAME"
  />
  <ScreenreaderText>
    NAME
  </ScreenreaderText>
</div>
`;

exports[`<Avatar /> with too many characters matches its snapshot 1`] = `
<div
  className="y-avatar y-avatar__size-medium"
>
  <StyledCustomizedPersonaCoin
    hidePersonaDetails={true}
    imageShouldFadeIn={false}
    size={12}
    text="First Last"
  />
  <ScreenreaderText>
    First Last
  </ScreenreaderText>
</div>
`;
